### PR TITLE
Fix and enhance NAT checksum update

### PR DESCRIPTION
**Pull request by ddiproietto**

### CODE REVIEW

File: core/utils/checksum.h [hunk 1]
```diff
@@ -298,17 +298,46 @@ static inline uint16_t CalculateIpv4TcpChecksum(const Ipv4Header &iph,
                                   ntohs(iph.length) - (iph.header_length << 2));
 }
 
+// Incremental checksum udpdate
```
Comment (barath) [inline core/utils/checksum.h]: update

File: core/utils/icmp.h [hunk 2]
```diff
@@ -5,8 +5,8 @@ namespace utils {
 
 // A basic ICMP header definition.
 struct[[gnu::packed]] IcmpHeader {
-  uint16_t type;      // ICMP packet type.
-  uint16_t code;      // ICMP packet type.
+  uint8_t type;       // ICMP packet type.
```
Comment (barath) [inline core/utils/icmp.h]: Given that this was off and we didn't catch it before, I wonder if there's an easy way for you to add a test / tests to this PR for the various checksum calculations?

File: core/utils/checksum.h (hunk 1 above)
Comment (ddiproietto) [inline core/utils/checksum.h]: fixed, thanks

File: core/utils/icmp.h (hunk 2 above)
Comment (ddiproietto) [inline core/utils/icmp.h]: good point, will do that

**Comment by barath** [comment]
Thanks, LGTM.  Will merge after CI finishes.

**Comment by barath** [comment]
photo 110

**Pull request merged**